### PR TITLE
moved location_id to link table

### PR DESCRIPTION
**Pull request by anurag-ks**
This is for my task - https://www.google-melange.com/gci/task/view/google/gci2014/5842417454415872

### CODE REVIEW

File: modules/s3db/doc.py [hunk 1]
```diff
@@ -226,7 +235,21 @@ def model(self):
...
                                       )
-
+        self.add_components(tablename,
+                            doc_document_location={"link": "doc_document_location",
+                                                   "key": "document_id",
+                                                   "joinby": "location_id"})
+        # ---------------------------------------------------------------------
+        # Document <> Location Link Table
+        #
+        tablename = "doc_document_location"
```
Comment (hitesh96db) [inline modules/s3db/doc.py:246]: tablename should be added to 'names' as well.

**Comment by hitesh96db** [comment]
apart from those minor changes :+1:

**Comment by anurag-ks** [comment]
@hitesh96db done

**Comment by anurag-ks** [comment]
@hitesh96db please review

**Pull request merged**